geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/GeoMesaMetadata.scala

```diff
@@ -384,5 +384,5 @@ object MultiRowAccumuloMetadata extends AccumuloMetadataRow {
 
   override def getRange(typeName: String): Range = Range.prefix(s"$typeName~")
 
-  override def getTypeName(row: Text): String = Text.decode(row.getBytes, 0, row.find("~"))
+  override def getTypeName(row: Text): String = Text.decode(row.getBytes, 0, row.toString.lastIndexOf("~"))
```

Comment:
Doh.  Looks like our life would be better if we could use \x00 as the separator...

Ok.... let's reconsider this...  typename is more configurable/change-able than a datasets field names...  What are our options:
1.  Introduce really, really complex code to sort out all the possibilities?  
2.  Not support typenames with ~'s in the name.  Throw an exception.
3.  Pull some kind of escaping?
4.  Others?